torch/quantization/fx/convert.py

```diff
@@ -335,11 +335,11 @@ def node_arg_is_quantized(node_arg: Any) -> bool:
         else:
             return False
 
-    def is_output_quantized(node: Node, obj: QuantizeHandler, qconfig: QConfigAny, modules: Dict[str, torch.nn.Module]) -> bool:
+    def is_output_quantized(node: Node, obj: QuantizeHandler, qconfig: QConfigAny, modules: Dict[str, torch.nn.Module], is_reference=False) -> bool:
```

Comment:
could we give an example why `is_reference` should matter when deciding whether the output is quantized?

Reply:
sure, this is because when is_reference is False, we would return float node when a certain dtype combination is not supported (since fbgemm/qnnpack only support certain dtype combinations), so the output may be float, but when is_reference is True, we support all dtype combinations so the output will always be quantized

Reply:
makes sense.  Is this the behavior we want?  Should we have a different flag to control this?  It seems unexpected that setting is_reference to True can change what is being quantized.

Reply:
hmm, this is an intermediate state, I think the alternative is that in case when the dtype combination is not supported when is_reference=False, we will produce the reference pattern. I think that is more aligned with what we'll have in the future.

Reply:
I think it's worth considering having `is_reference` only mean one thing (quantized ops are in reference format), and introduce a new setting to control what happens to ops whose quantization is not defined for fbgemm/qnnpack.  Otherwise, `is_reference` starts to have multiple meanings, it gets harder to explain, it starts to potentially change numerical outputs, and it grows in scope.

What are your thoughts?

Reply:
Yeah I understand that `is_reference` in the current state is not a very well defined flag. It's not the final state though, in terms of removing the ambiguity I think it might be easier just to remove this flag, we can do that gradually for each quantizehandler and then in the end remove it. I'm thinking of doing that, it's just not very high priority for me currently. Would we have someone that can help with this?

Reply:
so in the future, the semantics will be, we'll produce the pattern by default, if it's not supported by fbgemm, it will stay as a reference pattern and not be fused. does that sound good?

Reply:
>  It's not the final state though, to remove the ambiguity I think it might be easier just to remove this flag, we can do that gradually for each quantizehandler(one example is shown in next PR) and then in the end remove it.

makes sense if it's not the most urgent, adding a TODO sounds fine.  However, it seems like a p0 to me to get this done before rolling is_reference out to real customers.

Reply:
yeah, I think we should do it before the beta release

Reply:
> yeah, I think we should do it before the beta release

IMO this should be done before any internal customers use the code, so before prototype release :). We should discuss and come up with something which works for everyone.